com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/DebugUtility.java

```diff
@@ -118,6 +118,18 @@ public static IDebugSession launch(VirtualMachineManager vmManager,
         List<LaunchingConnector> connectors = vmManager.launchingConnectors();
         LaunchingConnector connector = connectors.get(0);
 
+        /** In the sun JDK 10, the first launching connector is com.sun.tools.jdi.RawCommandLineLauncher, which is not the one we want to use.
+         *  Add the logic to filter the right one from LaunchingConnector list.
+         *
+         *  This fix is only for the JDI implementation by JDK. Other JDI implementations (such as JDT) doesn't have the impact.
+         */
+        for (LaunchingConnector con : connectors) {
+            if (con.getClass().getName().equals("com.sun.tools.jdi.SunCommandLineLauncher")) {
```

Comment:
Make the string final static value as the other values.

Reply:
Nice. Fixed.